config/overrides.go

```diff
@@ -0,0 +1,172 @@
+
+import (
+	"regexp"
+	"strings"
+
+	tjconfig "github.com/crossplane/terrajet/pkg/config"
+	"github.com/crossplane/terrajet/pkg/types/name"
+	"github.com/pkg/errors"
+)
+
+// GroupKindCalculator returns the correct group and kind name for given TF
+// resource.
+type GroupKindCalculator func(resource string) (string, string)
+
+// GroupMap contains all overrides we'd like to make to the default group search.
+var GroupMap = map[string]GroupKindCalculator{
+	"rancher2_app":                         ReplaceGroupWords("app", 0),
+	"rancher2_app_v2":                      ReplaceGroupWords("appv2", 0),
+	"rancher2_auth_config_activedirectory": ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_adfs":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_azuread":         ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_freeipa":         ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_github":          ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_keycloak":        ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_okta":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_openldap":        ReplaceGroupWords("authconfig", 2),
+	"rancher2_auth_config_ping":            ReplaceGroupWords("authconfig", 2),
+	"rancher2_catalog":                     ReplaceGroupWords("catalog", 0),
+	"rancher2_catalog_v2":                  ReplaceGroupWords("catalogv2", 0),
+	"rancher2_certificate":                 ReplaceGroupWords("project", 0),
+	"rancher2_cloud_credential":            ReplaceGroupWords("", 0),
+	"rancher2_cluster":                     ReplaceGroupWords("cluster", 0),
+	"rancher2_cluster_v2":                  ReplaceGroupWords("clusterv2", 0),
+	"rancher2_config_map_v2":               ReplaceGroupWords("cluster", 0),
+	"rancher2_etcd_backup":                 ReplaceGroupWords("cluster", 0),
+	"rancher2_feature":                     ReplaceGroupWords("", 0),
+	"rancher2_machine_config_v2":           ReplaceGroupWords("", 0),
+	"rancher2_multi_cluster_app":           ReplaceGroupWords("appv2", 0),
+	"rancher2_namespace":                   ReplaceGroupWords("project", 0),
+	"rancher2_node_driver":                 ReplaceGroupWords("node", 0),
+	"rancher2_node_pool":                   ReplaceGroupWords("node", 0),
+	"rancher2_node_template":               ReplaceGroupWords("node", 0),
+	"rancher2_notifier":                    ReplaceGroupWords("cluster", 0),
+	"rancher2_project":                     ReplaceGroupWords("project", 0),
+	"rancher2_registry":                    ReplaceGroupWords("project", 0),
+	"rancher2_role_template":               ReplaceGroupWords("", 0),
+	"rancher2_secret":                      ReplaceGroupWords("secret", 0),
+	"rancher2_secret_v2":                   ReplaceGroupWords("secretv2", 0),
+	"rancher2_setting":                     ReplaceGroupWords("", 0),
+	"rancher2_storage_class_v2":            ReplaceGroupWords("cluster", 0),
```

Comment:
How `cluster` is related to `storage_class` here?

Reply:
I probably not immediately getting the logic behind the grouping

Reply:
A storage class is cluster scoped, see terraform-provider-rancher2 docs:

> Provides a Rancher Storage Class v2 resource. This can be used to manage k8s storage classes for Rancher v2 clusters and retrieve their information.

and 

> [cluster_id](https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/storage_class_v2#cluster_id) - (Required/ForceNew) The cluster id of the storageClass V2 (string)

The Rancher api is not very clearly structured, like the cloud provider api's of aws or google. So once you start adding references you end up in a golang cyclic dependency nightmare.

As storage class is cluster scoped I opted for grouping it in 'cluster', but if we prefer another grouping, that is fine with me. A group 'storagev2' with a single resource 'StorageClass' might be an option?